File: Cafebook/CafebookModel/Model/ModelWeb/KhachHang/DangKyDto.cs
namespace CafebookModel.Model.ModelWeb.KhachHang
{
    public class DangKyRequestDto
    {
        public string Email { get; set; } = string.Empty;
        public string SoDienThoai { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
    }

    public class VerifyOtpRequestDto
    {
        public int TempId { get; set; }
        public string Email { get; set; } = string.Empty;
        public string SoDienThoai { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string OtpCode { get; set; } = string.Empty;
    }

    public class DangKyResponseDto
    {
        public bool Success { get; set; }
        public string? Message { get; set; }

        public bool RequireOtp { get; set; }
        public bool IsOfficialAccount { get; set; } 
        public int? TempId { get; set; }
        public string? TempEmail { get; set; }
        public string? TempPhone { get; set; }

        public DangKyKhachHangDto? KhachHangData { get; set; }
        public string? Token { get; set; }
    }

    public class DangKyKhachHangDto
    {
        public int IdKhachHang { get; set; }
        public string HoTen { get; set; } = string.Empty;
        public string? Email { get; set; }
        public string? SoDienThoai { get; set; }
        public string? TenDangNhap { get; set; }
        public string? AnhDaiDienUrl { get; set; }
    }
}

File: Cafebook/ADRCafebookApi/app/src/main/java/com/example/cafebook/models/DangKyDto.java
package com.example.cafebook.models;
import com.google.gson.annotations.SerializedName;

public class DangKyDto {
    public static class Request {
        public String email;
        public String soDienThoai;
        public String password;

        public Request(String email, String soDienThoai, String password) {
            this.email = email;
            this.soDienThoai = soDienThoai;
            this.password = password;
        }
    }

    public static class Response {
        @SerializedName("success") public boolean success;
        @SerializedName("message") public String message;
        @SerializedName("requireOtp") public boolean requireOtp;
        @SerializedName("isOfficialAccount") public boolean isOfficialAccount;
        @SerializedName("tempId") public int tempId;
        @SerializedName("tempEmail") public String tempEmail;
        @SerializedName("tempPhone") public String tempPhone;
    }
}
